skrf/calibration/tests/test_calibration.py

```diff
@@ -715,16 +716,17 @@ def test_z0(self):
         self.assertTrue(max(npy.abs(self.rlgc.z0-self.cal.z0)) < 1e-3)
 
     def test_shift(self):
-        self.assertTrue(self.cal.apply_cal(self.measured[3]) == self.wg.thru())
-
+        npy.testing.assert_allclose(self.cal.apply_cal(self.measured[3]).s, 
+                                    self.wg.thru().s, atol=1e-3)
+        
     def test_shift2(self):
         feed = self.rlgc.line(50,'um')
         dut = self.wg.random(n_ports=2)
         #Thrus convert the port impedances to 50 ohm
         dut_feed = self.wg.thru()**feed**dut**feed**self.wg.thru()
         dut_meas = self.measure(dut_feed)
-        self.assertTrue(self.cal.apply_cal(dut_meas) == dut)
-
+        npy.testing.assert_allclose(self.cal.apply_cal(dut_meas).s, 
+                                    dut.s, atol=1.1e-3)
```

Comment:
do we know what the tol was before?  1e-3  is large. , is that the limit of current calcs, or am i mis-interpreting this?

Reply:
yes that was not normal. I found that the calibration methods also had normalization functions, which were interfering... Once corrected, the precision is the same as before (<1e-6 (ZERO) when equalling two network).

Reply:
I even put the absolute tolerance at 1e-11 and it passes.

Reply:
hum, it's more complicated. There are difference between previous impementation and the new pseudo-wave implementation, which is the one that should be used for calibrations. In particular, the `NISTMultilineTRLTest2` test case shows differences. 

Previous implementation of the `s2z` is 
```
npy.einsum('ijj->ij', sqrtz0)[...] = npy.sqrt(z0)
z =  sqrtz0 @ npy.linalg.inv(Id - s) @ (Id + s) @ sqrtz0
```
while the new one for pseudo-waves is:
```
npy.einsum('ijj->ij', U)[...] = npy.sqrt(z0.real)/npy.abs(z0)
npy.einsum('ijj->ij', ZR)[...] = z0
USU = npy.linalg.inv(U) @ s @ U
z = npy.linalg.inv(Id - USU) @ (Id + USU) @ ZR
```
In theory, both equations are the same when the characteristic impedance are real. But numerically, not. Playing with the precision of the numpy array helps in increasing the precision, but is not sufficient.

Reply:
maybe you could just test for imaginary part in z0 ,  and then do the numerically more stable thing if its less than tol.

Reply:
In fact, I was not correct. The two tests in `NISTMultilineTRLTest2` are "failing" (lower precision than before) _because_ they use a complex characteristic impedance. 

So, either I allow a lower precision (1E-4) for this test, or I keep the old implementation as a 3rd choice of `s_def`...

Reply:
so what is the difference in precision for the two implementations?  1e-4 vs 1e?

Reply:
vs 1e-6 (the default precision when doing `ntw1 == ntw2`).

Reply:
At the end, I kept using the previous implementation for calibration in `EightTerm.renormalize`, mostly for compatibility reason.